### PR TITLE
2113 comparative polydispersity measure

### PR DESCRIPTION
**Pull request by lucas-wilkins**
Implemented additional polydispersity measure: @smk78 to check this is what he wanted.

This PR is up to date with the recent PRs #2126 and #2125

### CODE REVIEW

**Review by smk78 — APPROVED** [review]
Functionality review on W10/x64.

Additional parameter value looks good!

The only thing is that, strictly-speaking, the existing parameter should be attributed to 'Eeckhaut' and not 'Ryan'. Can we make this change, please. I realise it might affect the tidy alignment of the boxes...

**Comment by wpotrzebowski** [comment]
@rozyczko will take a look on the code

File: src/sas/qtgui/Perspectives/Corfunc/UI/CorfuncPanel.ui [hunk 1]
```diff
@@ -7,7 +7,7 @@
     <x>0</x>
     <y>0</y>
     <width>762</width>
-    <height>1182</height>
+    <height>1228</height>
```
Comment (rozyczko) [inline src/sas/qtgui/Perspectives/Corfunc/UI/CorfuncPanel.ui:10]: will this not overflow the canvas at lower resolutions?
This has been a rather common issue with the corfunc widget, so we might want to start thinking about adding tabs and changing the layouts... (obviously not here!)

**Review by rozyczko — APPROVED** [review]
Considerable improvement to code quality.

Did not check the functionality.

**Review by krzywon — CHANGES_REQUESTED** [review]
Serialization fails at the `self.txtPolyStribeck()` I noted in one of my comments. Everything else looks good and would have been approved otherwise.

File: src/sas/qtgui/Perspectives/Corfunc/UI/CorfuncPanel.ui (hunk 1 above)
Comment (lucas-wilkins) [reply src/sas/qtgui/Perspectives/Corfunc/UI/CorfuncPanel.ui:10]: yeah, there's more fields to come too!

Comment (lucas-wilkins) [reply src/sas/qtgui/Perspectives/Corfunc/UI/CorfuncPanel.ui:10]: I will have a look at this with #2110 or something similar

File: src/sas/qtgui/Perspectives/Corfunc/CorfuncPerspective.py [hunk 2]
```diff
@@ -663,7 +671,8 @@ def getState(self):
             'avg_hard_block_thick': self.txtAvgHardBlock.text(),
             'avg_soft_block_thick': self.txtAvgSoftBlock.text(),
             'local_crystalinity': self.txtLocalCrystal.text(),
-            'polydispersity': self.txtPolydisp.text(),
+            'polydispersity': self.txtPolyRyan.text(),
+            'polydispersity_stribeck': self.txtPolyStribeck(),
```
[the remark this answers is not in the document]
Comment (lucas-wilkins) [reply src/sas/qtgui/Perspectives/Corfunc/CorfuncPerspective.py:675]: don't know how that squeezed through!

File: test/corfunc/utest_corfunc.py [hunk 3]
```diff
@@ -88,11 +88,7 @@ def transform_callback(self, transforms):
     def extract_params(self):
         params = self.calculator.extract_parameters(self.transformation[0])
 
-        # Removing the tests below, will be scrapping the dict for a dataclass in dependant PR very soon
-        #self.assertIsNotNone(params)
-        #self.assertEqual(len(params), 6)
-
-        self.assertLess(abs(params['max']-75), 2.5) # L_p ~= 75
+        self.assertLess(abs(params.long_period-75), 2.5) # L_p ~= 75
```
[the remark this answers is not in the document]
Comment (lucas-wilkins) [reply test/corfunc/utest_corfunc.py:91]: Yeah, tests are good. We should probably lock in the calculation behaviour at some point soon.

File: src/sas/qtgui/Perspectives/Corfunc/CorfuncPerspective.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (lucas-wilkins) [reply src/sas/qtgui/Perspectives/Corfunc/CorfuncPerspective.py]: Done

**Review by krzywon — APPROVED** [review]

**Pull request merged**